OrbitCaptureClient/CaptureClient.cpp

```diff
@@ -85,14 +106,40 @@ void CaptureClient::Capture(
   FinishCapture();
 }
 
-void CaptureClient::StopCapture() {
+bool CaptureClient::StopCapture() {
+  absl::MutexLock lock(&state_mutex_);
+  if (state_ == State::kStarting) {
+    // Block and wait until the state is not kStarting
+    bool (*IsNotStarting)(State * state) = [](State* state) {
+      return *state != State::kStarting;
+    };
+    state_mutex_.Await(absl::Condition(IsNotStarting, &state_));
+  }
+
+  if (state_ != State::kStarted) {
+    LOG("StopCapture ignored, because it is already stopping or stopped");
+    return false;
+  }
+
   CHECK(reader_writer_ != nullptr);
 
   if (!reader_writer_->WritesDone()) {
-    ERROR("Finishing writing on Capture's gRPC stream");
-    FinishCapture();
+    // It looks like there is no way to recover from that
+    // we can't safely reset reade_writer_ here because it
+    // will likely lead to null pointer dereference on the
+    // capture thread on reader_writer_->Read(). And no good
```

Comment:
Isn't there a way to protect from this by somehow checking if `reader_writer_` has become a null pointer? This is a regression and we should really avoid crashing on purpose.

Reply:
why is it a regression?

Reply:
So you mean that the null pointer dereferencing already happened before? Either way, is there really no way we can protect against that instead of just crashing?

Reply:
Added forced stop flag...